File: AlgorithmReview/Programmers/Lv2/프로세스.swift
//
//  프로세스.swift
//  AlgorithmReview
//
//  Created by 박소윤 on 2023/05/05.
//

import Foundation

func solution(_ priorities:[Int], _ location:Int) -> Int {
    
    var sorted = priorities.sorted(by: <)
    var priorities = Array(priorities.enumerated())
    var count = 1
    
    while true {
        let pop = priorities.removeFirst()
        if sorted.last! > pop.element {
            priorities.append(pop)
        } else {
            if pop.offset == location {
                return count
            }
            sorted.removeLast()
            count += 1
        }
    }
    
    return count
}

File: Python/프로세스.py
def solution(priorities, location):
    
    data = [(i,p) for (i,p) in enumerate(priorities)]
    priorities.sort()
    
    result = 0
    while len(data) != 0:
        
        pop = data.pop(0)
        
        if priorities[-1] == pop[1]:
            result += 1
            if pop[0] == location:
                break
            else:
                priorities.pop()
                continue
        else:
            data.append(pop)
                
    return result
